builtin/credential/approle/path_tidy_user_id.go

```diff
@@ -148,6 +146,13 @@ func (b *backend) tidySecretID(ctx context.Context, req *logical.Request) (*logi
 				return nil
 			}
 
+			// Wait to get a lock for each of the secretIDs to make sure we avoid a race conditon
+			// where the accessor has been written but not yet the secretIDHMAC
+			for _, lock := range b.secretIDLocks {
+				lock.Lock()
+				lock.Unlock()
```

Comment:
Hmm, this still doesn't seem right to me. I've never seen locks grabbed before without doing some work before releasing them. I suspect that in a racy production environment this would not prevent races, though it would slow the code execution.

Reply:
> I suspect that in a racy production environment this would not prevent races, though it would slow the code execution.

Are you referring to the race condition I'm describing in https://github.com/hashicorp/vault/pull/6252#discussion_r263463273 ? Or is there another one that you can see that could potentially cause issues?

@jefferai are you able to chime in? I don't think either of us are 100% sure if my proposal here is safe and you are the original author of the dangling accessor cleanup code.